webgetApp/src/webPOSTRequest.cpp

```diff
@@ -0,0 +1,72 @@
+/** @file charToStringWaveform.c
+ *  @author Freddie Akeroyd, STFC (freddie.akeroyd@stfc.ac.uk)
+ *  @ingroup asub_functions
+ *
+ *  Copy a CHAR waveform record into a STRING waveform record. If this is done by
+ *  a normal CAPUT the character byte codes are not preserved
+ *
+ *  It expect the A input to be the waveform data and B to be "NORD" (number of elements)
+ *  it write its output to VALA
+ */
+#include <string>
+
+#include <registryFunction.h>
+#include <aSubRecord.h>
+#include <menuFtype.h>
+#include <errlog.h>
+
+// CURL
+#include <curl/curl.h>
+#include <curl/easy.h>
+
+#include "webUtils.h"
+
+// nmemb can be 0 and data may not be NULL terminated
+static size_t write_callback(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+	aSubRecord *prec = (aSubRecord *)userdata;
+	std::string message(ptr, nmemb * size);
+	errlogSevPrintf(errlogInfo, "%s: %s\n", prec->name, message.c_str());
+	return nmemb;
+}
+
+#include <epicsExport.h>
```

Comment:
Can't remember why it is there, as it switches sense of export/import directives  it sometimes needs adding at a strang point but in this case it can move up. i may have had something else above it at some point.